crates/ruma-events/src/secret/request.rs

```diff
@@ -0,0 +1,52 @@
+//! Types for the *m.secret.request* event.
+
+use ruma_events_macros::EventContent;
+use ruma_identifiers::DeviceIdBox;
+use ruma_serde::StringEnum;
+use serde::{Deserialize, Serialize};
+
+use crate::ToDeviceEvent;
+
+/// Event sent by a client to request a secret from another device or to cancel a previous request.
+///
+/// It is sent as an unencrypted to-device event.
+pub type SecretRequestEvent = ToDeviceEvent<SecretRequestEventContent>;
+
+/// The payload for SecretRequestEvent
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+#[ruma_event(type = "m.secret.request", kind = ToDevice)]
+pub struct SecretRequestEventContent {
+    /// The name of the secret that is being requested.
+    ///
+    /// Required if `action` is `request`.
+    pub name: Option<String>,
```

Comment:
I think it would make sense for this to be part of the `RequestAction` enum so we encode that the action of type `request` requires a name.

We might also want to consider using an enum for this instead there are some known names that are used, e.g. for the private cross signing keys:
> The cross-signing private keys can be stored on the server or shared with other devices using the Secrets module. When doing so, the master, user-signing, and self-signing keys are identified using the names m.cross_signing.master, m.cross_signing.user_signing, and m.cross_signing.self_signing

For the backup key:

> The recovery key can also be stored on the server or shared with other devices using the Secrets module. When doing so, it is identified using the name m.megolm_backup.v1, and the key is base64-encoded before being encrypted.

Reply:
I tried applying your suggestions with the new `RequestAction` and `SecretName` types. Let me know if it's in line with what you expect, or if there are other known names that I missed and could add.